### PR TITLE
Specialize output for timeout error

### PR DESCRIPTION
**Pull request by kriskowal**
r @ShanniLi @Raynos

### CODE REVIEW

File: logger.js [hunk 1]
```diff
@@ -65,6 +65,16 @@ Logger.prototype.error = function error(err) {
             fullType: err.fullType
         }));
 
+    } else if (err.type === 'tchannel.request.timeout') {
```
Comment (ShanniLi) [inline logger.js]: timeout can be an error frame, right? i.e., err.isErrorFrame == true

Comment (kriskowal) [inline logger.js]: Evidently not in all cases. It depends on whether TChannel produces the timeout locally, or whether the timeout comes from a peer over the wire.

Comment (ShanniLi) [inline logger.js]: This line will take the error frame case:
https://github.com/uber/tcurl/pull/98/files#diff-7f28009834c7b72b5b0080bc0487ccddL45

Comment (kriskowal) [inline logger.js]: Yes, it does. Whether the timeout happens in the local TChannel client or in a remote TChannel client is random. If it is local, we do not get an error frame.

**Comment by ShanniLi** [comment]
lgtm

**Comment by kriskowal** [comment]
Closing in favor of simpler #99

**Pull request closed without merging**